cljfmt/src/cljfmt/core.cljc

```diff
@@ -297,6 +297,31 @@
 (defn remove-trailing-whitespace [form]
   (transform form edit-all trailing-whitespace? zip/remove))
 
+
+(defn sort-requires
+  [require-zloc]
+  (z/replace
```

Comment:
Can you explain what you're doing with this function? Also how does it handle changing the indent level?

Reply:
This bit is a little tricky. First it collect all the vectors within the `(:require..)`, sorts them, then effectively plays them back in sorted order in the loop, replacing only the vector nodes. This leaves all the original whitepspace nodes in place.

Reply:
That's an interesting approach. It would mean that comments wouldn't follow their associated clause, but comments are rare in the `ns` declaration anyway.

You seem to be looking for vectors, but namespaces in `require` can just be symbols. Perhaps instead look for non-whitespace tokens?

Maybe you want something more like:

```clojure
(let [nodes  (partition-by whitespace? children)
      spaces (take-nth 2 nodes)
      forms  (take-nth 2 (rest nodes))]
  (->> forms (sort-by sexpr) (interleave spaces) (apply concat)))
```

Or even factor out the unravelling:

```clojure
(defn unravel [n coll]
  (map #(take-nth n (drop % coll)) (range n)))

;; ...

(let [[spaces forms] (->> children (partition-by whitespace?) (unravel 2))]
  (->> forms (sort-by sexpr) (interleave spaces) (apply concat)))
```

Obviously that's pseudo-ish code, but should give you a rough idea of what I mean.

Reply:
I think it'd be cool if it worked with comments too, so I'll take a look at that. We can't rely on the same hard-coded unraveling then though.